src/figuresClasses.ts

```diff
@@ -1,19 +1,76 @@
-export interface Figure {
+type Shape = 'triangle' | 'circle' | 'rectangle';
+type Color = 'red' | 'green' | 'blue';
 
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea(): number;
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  shape: Shape = 'triangle';
 
-}
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('One of the sides is not a positive number');
+    }
+
+    if (Math.max(a, b, c) >= Math.min(a, b) + Math.min(b, c)) {
+      throw new Error(
+        `sides ${this.a}, ${this.b} and ${this.c} can't form a triangle`,
+      );
+    }
+  }
 
-export class Circle {
+  getArea(): number {
+    const semiPer = (this.a + this.b + this.c) / 2;
 
+    return Math.floor(Math.sqrt(
+      semiPer * (semiPer - this.a) * (semiPer - this.b) * (semiPer - this.c),
+    ) * 100) / 100;
+  }
 }
 
-export class Rectangle {
+export class Circle implements Figure {
+  shape: Shape = 'circle';
 
+  constructor(
+    public color: Color,
+    public radius: number,
+  ) {
+    if (this.radius <= 0) {
+      throw new Error('Radius is not a positive number');
+    }
```

Comment:
you check input value for figures sizes in each class
will be better to create separate function for this purpose and call this function in each class